cron/pubsub/subscriber_gcs.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pubsub
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	pubsub "cloud.google.com/go/pubsub/apiv1"
+	pubsubpb "google.golang.org/genproto/googleapis/pubsub/v1"
+
+	"github.com/ossf/scorecard/cron/data"
+)
+
+const (
+	maxMessagesToPull         = 1
+	ackDeadlineExtensionInSec = 600
+	gracePeriodInSec          = 60
+	gcpPubsubPrefix           = "gcppubsub://"
+)
+
+type gcsSubscriber struct {
+	ctx             context.Context
+	done            chan bool
+	client          *pubsub.SubscriberClient
+	pullRequest     *pubsubpb.PullRequest
+	subscriptionURL string
+	recvdAckIDs     []string
```

Comment:
Any reason why this needs to be a slice? The way the code is structured right now is pretty tied to the a maximum of 1 message at a time.

Reply:
True. Updated.